src/shape.rs

```diff
@@ -1437,14 +1463,19 @@ impl ShapeLine {
                 }
             }
 
-            layout_lines.push(LayoutLine {
-                w: if align != Align::Justified {
-                    visual_line.w
-                } else if self.rtl {
-                    start_x - x
+            let current_line_width = if align != Align::Justified {
+                if index == 0 {
+                    visual_line.w - first_line_indent
                 } else {
-                    x
-                },
+                    visual_line.w
+                }
```

Comment:
uber-nit: just to reduce nesting a bit more, could also in-line `visual_line.w - if index == 0 { first_line_indent } else { 0 }`. Ultimately, not sure if this is any cleaner haha 😅 . If we did in-line, can do this too
```
else {
    if self.rtl { start_x - x } else { x }
}
```

Could also move it into its own variable but technically doing some extra work there (though it doesn't really matter), if we're in the justified case.

Anyhow, open to leave as-is too - just was a bit tricky to initially read due to the nesting.

Reply:
This is indeed a bit difficult to read. I've changed the structure a bit (while still trying to conform to `cargo fmt`'s preferences) -- is it easier to parse now?

Reply:
Easier to read! Thanks - wondering if we can also further simplify to this (basically take out the second block's nested if-else and bring it to top-level?):
```
            let current_line_width = if align != Align::Justified {
                visual_line.w - if first_line { first_line_indent } else { 0. }
            } else if self.rtl {
                start_x - x
            } else {
                x
            };
```